appmap/_implementation/event.py

```diff
@@ -29,9 +31,16 @@ def next_thread_id(cls):
     @classmethod
     def get_thread_id(cls):
         tls = utils.appmap_tls()
-        if 'thread_id' not in tls:
-            tls['thread_id'] = cls.next_thread_id()
-        return tls['thread_id']
+        try:
+            thread_id = tls['thread_id']
+        except KeyError:
+            thread_id = tls['thread_id'] = cls.next_thread_id()
+        return thread_id
```

Comment:
try this
`return tls.setdefault('thread_id', cls.next_thread_id())`

Reply:
`>>> d = {}
>>> d.setdefault('a', 2)
2
>>> d
{'a': 2}
>>> d['a'] = 3
>>> d.setdefault('a', 2)
3`

Reply:
I don't think we can do that. `cls.next_thread_id()` should only be called if `thread_id` doesn't exist in `tls`.

Reply:
ahh right. It will populate next_thread_id even if value is found. my bad.

Reply:
few alternate oneliners

```
>>> def foo():
...     print('foo')
...     return 2
...
```

```
>>> d = {}
>>> d['a'] if 'a' in d else d.setdefault('a', foo())
foo
2
>>> d['a'] if 'a' in d else d.setdefault('a', foo())
2
```

```
>>> d = {}
>>> d.get('a') or d.setdefault('a', foo())
foo
2
>>> d.get('a') or d.setdefault('a', foo())
2
```

Reply:
I believe performance will be important here, and the `try...except` idiom seems to be used quite a bit. It seems to me that your second suggestion has all the benefits of the `try...except` construct, but reads better. I'll switch to that.